schema/equals_test.go

```diff
@@ -60,86 +60,122 @@ func TestEquals(t *testing.T) {
 	// The "copy known fields" section of these function is to break if folks
 	// add new fields without fixing the Equals function and this test.
 	funcs := []interface{}{
-		func(x Schema) bool {
+		func(x *Schema) bool {
+			if x == nil {
+				// skip nil checks as the method can not be called on nil
+				return true
+			}
 			if !x.Equals(x) {
 				return false
 			}
-			var y Schema
```

Comment:
I'd revert all of the changes in this file, and simply change the "Equals" call to pass `&x` or `&y`.

Reply:
just did. seems reasonable. just need to keep the changes for Map and Schema because of their locks.